lib/orbit/evented.js

```diff
@@ -191,72 +191,36 @@ var Evented = {
       return listeners;
     },
 
-    resolve: function(eventNames) {
-      var listeners = this.listeners(eventNames),
-          args = Array.prototype.slice.call(arguments, 1);
-
-      return new Orbit.Promise(function(resolve, reject) {
-        var resolveEach = function() {
-          if (listeners.length === 0) {
-            reject();
-          } else {
-            var listener = listeners.shift();
-            var response = listener[0].apply(listener[1], args);
-
-            if (response) {
-              response.then(
-                function(success) {
-                  resolve(success);
-                },
-                function(error) {
-                  resolveEach();
-                }
-              );
-            } else {
-              resolveEach();
-            }
-          }
-        };
-
-        resolveEach();
-      });
+    resolve(eventNames, ...args) {
```

Comment:
Yeah, I'm not sure `resolve` is the best name actually, at this point it's competing with Promise.resolve(). `find` or `detect` would clearer for me (ok, I'm a rubyist). @dgeb what do you reckon?